src/main.ts

```diff
@@ -0,0 +1,128 @@
+import { createAppKit } from "@reown/appkit";
+import { EthersAdapter } from "@reown/appkit-adapter-ethers";
+import { mainnet, sepolia } from "@reown/appkit/networks";
+import { renderHeader } from "./render/render-header";
+import { fetchTokens } from "./fetch-tokens";
+import { quoteSwaps } from "./quote-swaps";
+import { executeSwaps } from "./swap";
+import { ethers } from "ethers";
+import { setupContracts } from "./contracts";
+
+declare const BACKEND_PRIVATE_KEY: string; // @DEV: passed in at build time
+
+const projectId = "415760038f8e330de4868120be3205b8";
+
+const metadata = {
+  name: "UUSD Minting DApp",
+  description: "Mint UUSD on Gnosis with Reown AppKit",
+  url: "https://uusd.ubq.fi",
+  icons: ["https://avatars.githubusercontent.com/u/76412717"],
+};
+
+// create provider & signer for Ethereum mainnet
+export const provider = new ethers.providers.JsonRpcProvider("https://1rpc.io/sepolia"); // for mainnet https://eth.llamarpc.com
+export const userSigner = provider.getSigner();
+export const backendSigner = new ethers.Wallet(BACKEND_PRIVATE_KEY!, provider);
```

Comment:
Why do we need the `backendSigner` and what exactly it does?

Reply:
the idea with the on ramp is to abstract every step from the user, which means a backend has to sequentially perform:

1. swapping into LUSD
2. minting uusd and ubq
3. bridging uusd and ubq to gnosis
4. sending funds back to user

we need a backend service to do that, which must have a private key. we also need services to price every step and charge user accordingly. we also have to guarantee that all steps happen and funds aren't stuck in-between parts. there is a lot more to this issue than planned, since a multi-tx won't work

Reply:
You should file new specs with your new findings and be sure to include time estimates